staker/token_register.gno

```diff
@@ -21,144 +22,114 @@ type GRC20Interface interface {
 }
 
 var (
-	registered = make(map[string]GRC20Interface)
+	registered = avl.NewTree()
 	locked     = false // mutex
 )
 
 // GetRegisteredTokens returns a list of all registered tokens
 func GetRegisteredTokens() []string {
-	tokens := make([]string, 0, len(registered))
-	for k := range registered {
-		tokens = append(tokens, k)
-	}
+	tokens := make([]string, 0, registered.Size())
+
+	registered.Iterate("", "", func(key string, value interface{}) bool {
+		tokens = append(tokens, key)
+		return false
+	})
+
 	return tokens
 }
 
 // RegisterGRC20Interface registers a GRC20 token interface
 func RegisterGRC20Interface(pkgPath string, igrc20 GRC20Interface) {
 	prevAddr := std.PrevRealm().Addr()
 	prevPath := std.PrevRealm().PkgPath()
-	if !(prevAddr == consts.TOKEN_REGISTER || prevPath == consts.INIT_REGISTER_PATH || strings.HasPrefix(prevPath, "gno.land/r/g1er355fkjksqpdtwmhf5penwa82p0rhqxkkyhk5")) {
+	if !(prevAddr == consts.TOKEN_REGISTER || prevPath == consts.INIT_REGISTER_PATH || strings.HasPrefix(prevPath, consts.TOKEN_REGISTER_NAMESPACE)) {
 		panic(addDetailToError(
```

Comment:
Avoid using panic for general errors. Return errors instead

Reply:
This and dealing with global variables go the same for your other ongoing refactor work, including opened PRs. Address them.